### PR TITLE
feat(commit_description_parser): take pr number instead

### PR DESCRIPTION
**Pull request by vdice**
And derive latest commit sha from pr number given
via appropriate GitHub API req.

Ref https://github.com/deis/workflow/pull/366
Closes https://github.com/deis/jenkins-jobs/issues/111

### CODE REVIEW

File: bash/scripts/commit_description_parser.sh [hunk 1]
```diff
@@ -10,25 +10,47 @@ main() {
...
   done
 }
 
+get-most-recent-pr-commit() {
+  repoName="${1}"
+  prNumber="${2}"
+
+  curl \
+  -sSL \
+  --user deis-admin:"\${GITHUB_ACCESS_TOKEN}" \
```
Comment (sgoings) [inline bash/scripts/commit_description_parser.sh:52]: Might want to move `deis-admin` to a variable.

File: bash/scripts/commit_description_parser.sh [hunk 2]
```diff
@@ -10,25 +10,47 @@ main() {
...
+  "https://api.github.com/repos/deis/${repoName}/pulls/${prNumber}/commits" \
+  | docker run -i --rm kamermans/jq '.[].sha' | tail -n1
```
Comment (sgoings) [inline bash/scripts/commit_description_parser.sh:54]: This seems like a bit of overhead just to run jq. Not a huge deal for now... but might we have better options for running jq quickly?

**Comment by sgoings** [comment]
Wooooo eeee. That bash test is advanced! 💯 

Tested manually: 👍

File: bash/scripts/commit_description_parser.sh [hunk 3]
```diff
@@ -10,25 +10,47 @@ main() {
...
-# to pass along to downstream job(s)...
+# commits to pass along to downstream job(s)...
 parse-commit-description() {
   description="${1}"
 
-  # Looks specifically for matches of '[rR]equires <repo>#<sha>',
-  # e.g., "requires builder#abc1234, Requires router#def5678"
-  reqs=$(echo "${description}" | grep -o "[Rr]equires [-a-z]*#[a-z0-9]*" | grep -o "[-a-z]*#[a-z0-9]*") || true
+  # Looks specifically for matches of '[rR]equires <repo>#<prNumber>',
+  # e.g., "requires builder#123, Requires router#567"
```
Comment (sgoings) [inline bash/scripts/commit_description_parser.sh:18]: What happened to `4`? 😢

File: bash/scripts/commit_description_parser.sh [hunk 4]
```diff
@@ -10,25 +10,47 @@ main() {
...
+  reqs=$(echo "${description}" | grep -o "[Rr]equires [-a-z]*#[0-9]*" | grep -o "[-a-z]*#[0-9]*") || true
```
Comment (sgoings) [inline bash/scripts/commit_description_parser.sh]: Slight regex edge case:

Would pass on:

`Requires controller#`

File: bash/scripts/commit_description_parser.sh [hunk 5]
```diff
@@ -10,25 +10,47 @@ main() {
...
 
-  # split on whitespace into array of '<repo>#<sha>' values
+  # split on whitespace into array of '<repo>#<prNumber>' values
   reqsArray=(${reqs// / })
 
-  # for each '<repo>#<sha>' value in array
+  # for each '<repo>#<prNumber>' value in array
   for i in "${reqsArray[@]}"; do
```
Comment (sgoings) [inline bash/scripts/commit_description_parser.sh:26]: Above this you should reset `i`:

```
local i
```

If you don't, `i` can contain something else... and you'd process it below.

File: bash/scripts/commit_description_parser.sh (hunk 2 above)
Comment (vdice) [inline bash/scripts/commit_description_parser.sh:54]: yea agreed, a bit 'heavy-handed' (though image only ~12MB).  Went this route b/c I knew ('linux') Jenkins slaves have docker, but not jq (yet).  Shall I create an issue to add `jq` to the Jenkins slaves so we can run directly -- and refactor this dependency out?

**Comment by sgoings** [comment]
LGTM-ing due to mostly nits. Fixup as you please.

File: bash/scripts/commit_description_parser.sh (hunk 2 above)
Comment (arschles) [inline bash/scripts/commit_description_parser.sh:54]: I'm sure you anticipated this comment from me... it seems like this suite of scripts is now complex enough to warrant building a Go program to replace (at least part of) it. Thoughts on doing that eventually?

Comment (vdice) [inline bash/scripts/commit_description_parser.sh:54]: @arschles mos def.  It's been on my mind as well.  We should brainstorm soon and generate a meta ticket w/ outline of transition.

**Pull request merged**